web3/main.py

```diff
@@ -125,3 +147,48 @@ def createBatch(self):
 
     def receive(self, requestid, timeout=0, keep=False):
         return self._requestManager.receive(requestid, timeout, keep)
+
+
+def hex_encode_abi_type(abi_type, value):
+    if is_array_type(abi_type):
+        sub_type = re.sub(r"\[[^]]*\]$", "", abi_type, 1)
+        return "".join([hex_encode_abi_type(sub_type, v) for v in value])
+    elif is_bool_type(abi_type):
+        return remove_0x_prefix(hex(value)).zfill(2)
+    elif is_uint_type(abi_type):
+        return positive_int_to_hex(value, size_of_type(abi_type))
+    elif is_int_type(abi_type):
+        bit_size = size_of_type(abi_type)
+        if value < 0:
+            return twos_compliment_of_negative(value, bit_size)
+        else:
+            return positive_int_to_hex(value, bit_size)
+    elif is_address_type(abi_type):
+        return remove_0x_prefix(value)
+    elif is_bytes_type(abi_type):
+        if is_0x_prefixed(value):
+            return remove_0x_prefix(value)
+        else:
+            return bytes_to_hex(value)
+    elif is_string_type(abi_type):
+        return bytes_to_hex(value)
+
+
+def positive_int_to_hex(value, bit_size=None):
+    hex_value = remove_0x_prefix(hex(value))
+    if bit_size:
+        return hex_value.zfill(int(bit_size / 4))
+    return ('0' * (len(hex_value) % 2)) + hex_value
+
+
+def twos_compliment_of_negative(value, bit_size):
+    value = (1 << bit_size) + value
+    hex_value = hex(value)
+    hex_value = hex_value[:-1] if hex_value[-1] == "L" else hex_value
+    return remove_0x_prefix(hex_value)
+
+
+def bytes_to_hex(byte_string):
+    if type(byte_string) == str:
+        byte_string = str.encode(byte_string)
+    return codecs.getencoder('hex')(byte_string)[0].decode("utf-8")
```

Comment:
Explicit encoding is better than implicit here, like `byte_string.encode('utf-8')` instead of `str.encode(byte_string)`

You could also leverage some of the tools from the utils library here, like: `return remove_0x_prefix(to_hex(byte_string))`